src/axom/slam/examples/lulesh2.0.3/lulesh-comm.cpp

```diff
@@ -1918,10 +1918,12 @@ namespace slamLulesh {
           cmsg * CACHE_COHERENCE_PAD_REAL];
       Index_t idx = dx * dy * (dz - 1);
       MPI_Wait(&domain.recvRequest[pmsg + emsg + cmsg], &status);
-      for (Index_t fi = 0; fi<xferFields; ++fi)
-      {
-        (domain.*fieldData[fi])(idx) = comBuf[fi];
-      }
+      (domain.*fieldData[0])(idx) = comBuf[0];
+      (domain.*fieldData[1])(idx) = comBuf[1];
+      (domain.*fieldData[2])(idx) = comBuf[2];
+      (domain.*fieldData[3])(idx) = comBuf[3];
+      (domain.*fieldData[4])(idx) = comBuf[4];
+      (domain.*fieldData[5])(idx) = comBuf[5];
```

Comment:
The sanitizer was throwing a `runtime error: index -3062 out of bounds for type '<unknown> [6]` for these loops. The actual values of `fi` and `idx` don't reflect an index out of bounds. Oddly, unrolling the loops removes the error...

Reply:
very strange -- please add a comment about why this was explicitly unrolled, and maybe an assert that `xferFields == 6`